magic-gds-drc/run_all_drc.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Copyright 2020 Efabless Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+-------------------------------------------------------------------------
+run_all_drc.py --- A script that will run run_standard_drc
+for all .gds files under the cells/ folder. Must be run from repository
+root.
+
+Usage: python3 run_all_drc.py --help
+
+Results:
+
+  Prints a report to standard output.
+	
+-------------------------------------------------------------------------
+"""
+
+import os
+import re
+import click
+import traceback
+import subprocess
+from typing import List, Tuple
+import concurrent.futures as fut
+
+acceptable_errors_list = []
+
+SCRIPT_DIR = os.path.realpath(os.path.dirname(__file__))
+STANDARD_DRC_SCRIPT = os.path.join(SCRIPT_DIR, "run_standard_drc.py")
+PDK_SUBSET = os.path.join(SCRIPT_DIR, "sky130A")
+
+DRCError = Tuple[str, List[str]]
+
+def parse_drc_report(report: str) -> List[DRCError]:
```

Comment:
The docstring description could be a doctest;
```python
    """
    Takes a magic report in the format:

    >>> report = '''\\
    ... Header
    ... 
    ... 
    ... Error 1
    ... Error
    ... Details
    ... 
    ... 
    ... Error 2
    ... Error
    ... Details
    ... 
    ... 
    ... Error N
    ... Error
    ... Details
    ... '''
    >>>  parse_drc_report(report)

    ....
    """
```

Reply:
Had to purify the function for that, slightly altering the signature, but I think it was worth it.